capsules/src/FXOS8700CQ.rs

```diff
@@ -0,0 +1,130 @@
+use core::cell::Cell;
+use kernel::{AppId, Callback, Driver};
+use kernel::common::take_cell::TakeCell;
+use kernel::hil::i2c::{I2CDevice, I2CClient, Error};
+
+pub static mut BUF: [u8; 6] = [0; 6];
+
+const DEFAULT_SCALE: u8 = 0x0;
+
+#[allow(dead_code)]
+enum Registers {
+    SensorStatus = 0x00,
+    OutXMSB = 0x01,
+    OutXLSB = 0x02,
+    OutYMSB = 0x03,
+    OutYLSB = 0x04,
+    OutZMSB = 0x05,
+    OutZLSB = 0x06,
+    XyzDataCfg = 0x0E,
+    WhoAmI = 0x0D,
+    CtrlReg1 = 0x2A,
+}
+
+#[derive(Clone,Copy,PartialEq)]
+enum State {
+    /// Sensor is disabled (but on)
+    Disabled,
+
+    /// Reading that sensor is present
+    Enabling,
+
+    /// Activate sensor for readings
+    Activating,
+
+    /// Deactivate sensor from readings
+    Deactivating(usize, usize, usize),
+
+    /// Reading accelerometer data
+    ReadingAcceleration,
+}
+
+pub struct FXOS8700CQ<'a> {
+    i2c: &'a I2CDevice,
+    state: Cell<State>,
+    buffer: TakeCell<&'static mut [u8]>,
+    callback: Cell<Option<Callback>>,
```

Comment:
Hrm... ok, this _should_ be virtualized, but none of the other sensors currently have virtualization, so maybe we should just do them altogether.